virttest/utils_kernel_module.py

```diff
@@ -0,0 +1,151 @@
+    """Class handling Linux kernel modules"""
+
+    def __init__(self, module_name, persistence_file=""):
+        """
+        Create kernel module handler. It backs up the current status
+        of the kernel module or get its current status from a backed up
+        file.
+
+        :param module_name:
+        :param persistence_file: absolute file path to store object in
+        """
+        self._module_name = module_name
+        self._was_loaded = None
+        self._loaded_config = ""
+        self._config_backup = ""
+        self._persistence_file = persistence_file
+        if os.path.isfile(self._persistence_file):
+            self._load()
+        else:
+            self._backup_config()
+
+    def load_module(self, force=True, params=""):
+        """
+        Load module with given parameters
+
+        :param force: Force load if currently not loaded. Default is True as
+        restore undoes loading.
+        :param params: Parameters to load with, 'key1=param1 key2=param2 ...'
+        """
+
+        if params == self._loaded_config:
+            logging.debug("Not reloading module, same parameters requested for"
+                          " module %s: %s" % (self._module_name, params))
+            return
+
+        if not force and not self._was_loaded:
+            logging.debug("Module %s isn't loaded. Set force=True to force"
+                          " loading." % self._module_name)
+        else:
+            cmd = ""
+            if self._was_loaded:
+                cmd += 'rmmod %s; ' % self._module_name
+            cmd += 'modprobe %s %s' % (self._module_name, params)
+            logging.debug("Loading module: %s" % cmd)
+            status, output = process.getstatusoutput(cmd, shell=True,
+                                                     ignore_status=True)
+            if status:
+                raise exceptions.TestError("Couldn't load module %s: %s" % (
+                    self._module_name, output
+                ))
+            else:
+                self._loaded_config = params
+
+        if self._persistence_file:
+            self.save()
+
+    def restore(self):
+        """Restore previous module state"""
+
+        if self._loaded_config:
+            if not self._was_loaded:
+                cmd = 'rmmod %s' % self._module_name
+            else:
+                cmd = 'rmmod %s; modprobe %s %s' % (self._module_name,
+                                                    self._module_name,
+                                                    self._config_backup)
+            logging.debug("Restoring module state: %s" % cmd)
+            status, output = process.getstatusoutput(cmd, shell=True,
+                                                     ignore_status=True)
+            if status:
+                raise exceptions.TestError("Couldn't restore module %s: %s" % (
+                    self._module_name, self._config_backup))
+
+        if self._persistence_file and \
```

Comment:
in avocado we use brackets and not escaped \ns (it's somewhere in code-style guide...)